core/server.go

```diff
@@ -391,7 +391,11 @@ func (m *RpcServer) ControlEnvironment(cxt context.Context, req *pb.ControlEnvir
 	td := eot.Sub(sot)
 
 	if err != nil {
-		env.Sm.SetState("ERROR")
+		err := env.TryTransition(environment.NewGoErrorTransition(m.state.taskman))
+		if err != nil {
+			log.WithField("partition", env.Id()).Warn("we could not transition gently to ERROR, thus forcing it.")
```

Comment:
Please rephrase and wrap: `Warnf("could not complete requested GO_ERROR transition, forcing move to ERROR: %w", err)`

Reply:
done, but used `err.Error()`, because `%w` may be used only with `Errorf`